src/js/binder.js

```diff
@@ -0,0 +1,128 @@
+/*
+
+    Add persistent bindings between a selector and a model value.  Changes are propagated between the two. See the
+    documentation for more details:
+
+    https://github.com/GPII/gpii-binder/
+
+    This code was originally written by Antranig Basman <amb26@ponder.org.uk> and with his advice was updated and
+    extended by Tony Atkins <tony@raisingthefloor.org>.
+
+*/
+/* global fluid, jQuery */
+(function () {
+    "use strict";
+    var gpii = fluid.registerNamespace("gpii");
+    fluid.registerNamespace("gpii.binder");
+
+    /**
+     *
+     * Try to evolve a given string into a typed variable or object.
+     *
+     * @param originalValue {String|Array} The value we will try to evolve.
+     * @returns {Object|String} An object if we were able to parse it, otherwise the original string.
+     *
+     */
+    gpii.binder.jsonOrString = function (originalValue) {
+        if (typeof originalValue === "string") {
+            // empty "value" elements in HTML should not result in a bunch of empty strings in our model
+            if (originalValue.length === 0) {
+                return null;
+            }
+            else {
+                try {
+                    var objectValue = JSON.parse(originalValue);
+                    return objectValue;
+                }
+                catch (e) {
+                    // Ignore the error and keep the string value.
+                    return originalValue;
+                }
+            }
+        }
+        else if (Array.isArray(originalValue)) {
+            return originalValue.map(gpii.binder.jsonOrString);
+        }
+        else {
+            return originalValue;
+        }
+    };
+
+    /**
+     * 
+     * A function to encode problematic values that will not work correctly with fluid.value -> jQuery.val().
+     * Adds special handling for boolean values, which would ordinarily be mangled.
+     * 
+     * @param originalData The original data, in its original form.
+     * @returns {Object|String} The original data, encoded as needed to work correctly with fluid.value.
+     */
+    gpii.binder.getSafeValue = function (originalData) {
+        if (Array.isArray(originalData)) {
+            return originalData.map(gpii.binder.getSafeValue);
+        }
+        else if (typeof originalData !== "string") {
+            return JSON.stringify(originalData);
+        }
+        else {
+            return originalData;
+        }
+    };
+
+    gpii.binder.applyChangeSafely = function (that, path, elementValue) {
```

Comment:
I added JSDocs here detailing why this function needs to exist and what it does.